File: solucoes/2020/nivel 1/fase 1/js/fissura.js
var M, F
var N
var matriz = []
var linha;
var novamatriz = []

scanf("%d%d", "M", "F")

    for (var i = 0; i < M; i++) {
        scanf("%s", "N")
        for (var j = 0; j < M; j++) {
            if (j % M === 0) {
                matriz.push([N])
            }
            else {
                matriz[matriz.length - 1].push(N)
            }
        }
    }

    printf(matriz.join('\n').replace(/,/g, ''))

    for (var k = 0; k < matriz.length; k++) {
        linha = matriz[k]
        for (var f = 0; f < linha.length; f++) {
            if (Number(linha[f]) <= F) {
                if (f % M === 0) {
                    novamatriz.push(['*'])
                }
                else {
                    novamatriz[novamatriz.length - 1].push('*')
                }
            }
            else {
                if (f % M === 0) {
                    novamatriz.push([linha[f]])
                }
                else {
                    novamatriz[novamatriz.length - 1].push(linha[f])
                }
            }
        }
    }
   // printf(novamatriz.join('\n').replace(/,/g, ''))

File: solucoes/2020/nivel 2/fase 1/py/fissura.py
N, F = map(int, input().split())
matriz=[]


for i in range(N):
  R = input()
  linha = []
  for i in range(N):
    linha.append(R[i])
  matriz.append(linha)    


checar = [[0,0]]
while len(checar):
  x,y = checar.pop()
  if(matriz[x][y] != "*" and F >= int(matriz[x][y])):
    matriz[x][y] = '*'
    if x < N-1:
      checar.append([x+1,y])
    if y < N-1:
      checar.append([x,y+1])
    if x > 0:
      checar.append([x-1,y])
    if y > 0:
      checar.append([x,y-1])

for i in range(len(matriz)):
  print(''.join(matriz[i]))